### ISSUE TITLE
Cannot render `InnerBlocks.Content` more than once 

### ISSUE THREAD

**Issue by michalczaplinski** [WordPress/block-hydration-experiments#23]
In our `wordpress-blocks.js` we are rendering the `<InnerBlocks.Content/>` twice: 

https://github.com/WordPress/block-hydration-experiments/blob/cb839e18c77a463ebe1c3b504f9f206c9faa81b0/src/gutenberg-packages/wordpress-blocks.js#L21

https://github.com/WordPress/block-hydration-experiments/blob/cb839e18c77a463ebe1c3b504f9f206c9faa81b0/src/gutenberg-packages/wordpress-blocks.js#L29

However, it's [not possible to render `<InnerBlocks/>` or `<InnerBlocks.Content/>` more than once](https://github.com/WordPress/gutenberg/blob/360511303434aa707c507ce621d37971bf08be32/packages/block-editor/src/components/inner-blocks/README.md) !

<a href="https://www.loom.com/share/96f545fad2ad4361b71603618d9570d9">
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/96f545fad2ad4361b71603618d9570d9-with-play.gif">
  </a>

https://www.loom.com/share/96f545fad2ad4361b71603618d9570d9

**Comment by luisherranz** [comment]
Thanks for reporting this, Michal 🙂

I'm going to quickly try if we can use `useInnerBlocksProps` (here in user-land) to solve this problem, but if we can't, we can mark it as `wontfix` because this shouldn't be a problem if/when we do this in Gutenberg itself.

**Comment by luisherranz** [comment]
It seems to work fine, so I've done a PR: https://github.com/WordPress/block-hydration-experiments/pull/26

**Comment by gziolo** [comment]
From the documentation:

> A block can render at most a single `InnerBlocks` and `InnerBlocks.Content` element in `edit` and `save` respectively. To create distinct arrangements of nested blocks, create a separate block type which renders its own `InnerBlocks` and assign as the sole allowedBlocks type.

It has been the limitation of the inner blocks so far. It might be tied to how the block parser works. @dmsnell or @mcsf should know better why we ended up with that approach.

**Comment by luisherranz** [comment]
To clarify: we don't pretend to render the inner blocks twice 🙂 

We are doing so in this repo because it didn't make sense to add the logic to distinguish between initially hidden inner blocks here. Once/if we propose this in Gutenberg, the inner blocks should be rendered only once, either inside a `<gutenberg-inner-blocks>` tag (if they are initially shown), or inside a `<template>` tag (if they are initially hidden). But never both.

**Comment by gziolo** [comment]
> To clarify: we don't pretend to render the inner blocks twice 🙂

Thank you for clarification. It makes sense 👍

Still, I’m curious to learn what is the full story of inner blocks 😃

**Comment by dmsnell** [comment]
> It might be tied to how the block parser works.

Nope 😄 

The parser supports an arbitrary arrangement of inner blocks scattered throughout a block because I have always felt it inevitable that we'd want to have multiple inner block regions within a single block.

So far the limit to a single location for a list of inner blocks is purely a decision (and the UI has been built around that decision) and we've gotten pretty far with this design.

It's possible today to include inner blocks in random places and it will parse properly but such a block will invalidate in the editor because of the limitation noted: it's not possible to `save()` those inner blocks in more than one place, to spread them around.

For the cases where we want different inner block regions I believe that the practice has generally been to crate a list-like container _as_ the inner blocks, and then limit programmatically what content is allowed inside that inner block container (on a per-block level).

**Comment by mcsf** [comment]
What Dennis said. :)

Look for the `innerContent` property of block nodes:

https://github.com/WordPress/gutenberg/blob/fb078c283e859e3cd49ae9354c78176fa4be851f/packages/block-serialization-default-parser/parser.php#L56-L68

**Comment by luisherranz** [comment]
> The parser supports an arbitrary arrangement of inner blocks scattered throughout a block

I've just realized that my fix in https://github.com/WordPress/block-hydration-experiments/pull/26 will lead to inner block duplication.

I'll open an issue and add a `wont-fix` because this shouldn't be a problem in the final implementation.

**Comment by gziolo** [comment]
Thank you @dmsnell and @mcsf for explaining where the limitation for inner blocks comes from. I guess it's a good compromise from the block editor's perspective, but it's also a bit limiting for how you might want express more complex blocks on the frontend. Do you know if it's documented in the block editor handbook? It would be worth clarifying for future generations 😄

**Comment by dmsnell** [comment]
I don't believe that the decision to not allow multiple inner block regions is not documented, but then again, what _is_ documented (I think) is that you have one place to put them. Maybe we can get some historical discussion of what led to this out there, but I have at least a tiny bit of hesitation to start documenting things that don't exist in the project.

We could have a note for "in case we ever want to do this" but I don't know where it would live. Any suggestions?

**Comment by gziolo** [comment]
> I don't believe that the decision to not allow multiple inner block regions is not documented, but then again, what is documented (I think) is that you have one place to put them. Maybe we can get some historical discussion of what led to this out there, but I have at least a tiny bit of hesitation to start documenting things that don't exist in the project.

I included the existing note from the `InnerBlocks` component in https://github.com/WordPress/block-hydration-experiments/issues/23#issuecomment-1134967125. It only mentions the limitation.  

User-facing documentation isn't the best place to include the note about current implementation. Maybe next to the component's definition? Still, it could be easy to miss there because of a few internal parts used.

https://github.com/WordPress/gutenberg/blob/bcbd38b8ae7221264873e2295abf333743069f7c/packages/block-editor/src/components/inner-blocks/index.js#L221-L224
